bip-vaults.mediawiki

```diff
@@ -0,0 +1,757 @@
+
+* '''withdrawal transaction''': spends <code>OP_UNVAULT</code> trigger inputs into a compatible set of final withdrawal outputs per the target hash, after the trigger inputs have matured per the spend delay. The only authorization for this spend (aside from the relative timelock) is the content hash of the withdrawal outputs.
+
+* '''recovery transaction''': spends one or more <code>OP_VAULT</code> or <code>OP_UNVAULT</code> inputs, which can be done at any time prior to withdrawal confirmation, to the prespecified recovery path. This transaction can optionally require a witness satisfying a specified ''recovery authorization'' script, an optional scriptPubKey gating the initiation of recovery. The use of recovery authorization has certain trade-offs discussed later.
+
+
+=== Parameters ===
+
+<code><recovery-params></code>
+
+The recovery parameters dictate both where funds can be swept to during a
+recovery, and what kind of authorization (if any) is needed to initiate a
+recovery. It is specified in the form 
+
+<source>
+<recovery sPK tagged hash (32 bytes)>[<optional auth. scriptPubKey ...>]
+</source>
+
+The first component commits to the destination that vault funds can be swept to
+at any point prior to the finalization of a withdrawal.
+
+The recovery scriptPubKey would usually correspond to a spending script that is
+inconvenient to exercise but highly secure.
+
+The second component, the recovery authorization scriptPubKey, is optional. It
+is a raw scriptPubKey that, if specified, must be satisfied to allow the input
+to be recovered. The benefit of using this parameter will be discussed later.
+If this component is not given, the de facto "authorization" is the reveal of
+the <code><recovery sPK tagged hash></code> preimage, i.e. the recovery path.
+
+Vaults which share the same recovery path can always be swept in batch operations,
+which is an important practical aspect of managing large numbers of vaults.
+
+<code><spend-delay></code>
+
+The spend delay dictates the duration of blocks or time which must
+elapse for the trigger <code>OP_UNVAULT</code> output to be claimable into the
+withdrawal target outputs. Encoded as the least significant 23 bits of a
+[https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki BIP-0068] style
+relative locktime.
+
+'''Trigger key'''
+
+The trigger key, committed to with <code><trigger-sPK-hash></code>, is used to
+authorize the ''trigger transaction'' - an on-chain declaration to attempt a
+withdrawal to a certain set of target outputs.
+
+This functions as the "normal" spending key, but if an attacker obtains access
+to this key, the outcome is not catastrophic: any withdrawal attempt can be
```

Comment:
I don't know if this statement is correct in the presence of unsolved pinnings vectors. E.g a bip125 rule #3 pinning where a malicious recovery transaction (without a secondary authorisation `scriptPubkey`) iiuc is absolute fee inflated to prevent confirmation of a recovery transaction. Though this one should be solved by nversion=3 I believe.